### PR TITLE
Update Matter SDK to v1.2.0.1

### PR DESCRIPTION
**Pull request by MonicaisHer**
The snap could be installed by running:
```
sudo snap install chip-tool --channel=latest/edge/pr23
```

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ parts:
   zap:
     plugin: nil
     build-environment:
-      - ZAP_VERSION: v2023.09.28
+      - ZAP_VERSION: v2023.10.24-nightly
```
Comment (farshidtz) [inline snap/snapcraft.yaml:36]: Is it necessary to use a release newer than v2023.09.28 (latest stable)?

Comment (MonicaisHer) [reply snap/snapcraft.yaml:36]: Yes, the Matter SDK 1.2.0.1 requires the minimal Zap version of v2023.10.24.

Comment (farshidtz) [reply snap/snapcraft.yaml:36]: That's problematic. Is it the same for v1.2.0.0?

Need to check why zap doesn't yet have a stable release compatible with Matter SDK v1.2. Might be a good idea to open an issue on the zap repo.

Comment (MonicaisHer) [reply snap/snapcraft.yaml:36]: For v1.2.0.0, it requires the [MIN_ZAP_VERSION = '2023.10.14'](https://github.com/project-chip/connectedhomeip/blob/v1.2.0.0/scripts/tools/zap/zap_execution.py#L26).

Comment (MonicaisHer) [reply snap/snapcraft.yaml:36]: issue opened on zap repo: https://github.com/project-chip/zap/issues/1209

**Review by farshidtz — APPROVED** [review]

**Pull request merged**